src/basic/verbs.c

```diff
@@ -23,12 +23,50 @@
 #include <stdbool.h>
 #include <stddef.h>
 
+#include "env-util.h"
 #include "log.h"
 #include "macro.h"
 #include "string-util.h"
 #include "verbs.h"
 #include "virt.h"
 
+/* Wraps running_in_chroot() which is used in various places,
+ * but also adds an environment variable check so external processes
+ * can reliably force this on.
+ */
+bool running_in_chroot_or_offline(void) {
+        int r;
+
+        /* We've had this condition check for a long time which basically
+         * checks for legacy chroot case like Fedora's
+         * "mock", which is used for package builds.  We don't want
+         * to try to start systemd services there, since without --new-chroot
+         * we don't even have systemd running, and even if we did, adding
+         * a concept of background daemons to builds would be an enormous change,
+         * requiring considering things like how the journal output is handled, etc.
+         * And there's really not a use case today for a build talking to a service.
+         */
+        r = running_in_chroot();
+        if (r < 0)
+                log_debug_errno(r, "running_in_chroot(): %m");
+        else if (r > 0)
+                return true;
+
+        /* Added to support use cases like rpm-ostree, where from %post
+         * scripts we only want to execute "preset", but not "start"/"restart"
+         * for example.
+         */
+        r = getenv_bool("SYSTEMD_OFFLINE");
```

Comment:
this check should be done first, so that it always wins if the env var is set. running_in_chroot() should only be used if the env var is not actually set, as an automatic fallback if nothing is explicitly configured.